tests/haddock/BUILD.bazel

```diff
@@ -56,8 +56,32 @@ haskell_doc(
     deps = [":haddock-lib-b"],
 )
 
+sh_test(
+    name = "data-dependency",
+    srcs = ["data_dependency.sh"],
+    args = ["$(rootpaths :haddock)"],
+    data = [":haddock"],
+    tags = [
+        # Fails in profiling mode due to missing haddocks for Deep.
+        "requires_dynamic",
```

Comment:
This is not a regression introduced with this PR but already the case on `master`. I.e. the following fails on master:
```
$ nix-shell --pure --run 'bazel test //tests/haddock/... -c dbg'
...
ERROR: /.../rules_haskell/tests/haddock/BUILD.bazel:18:1: HaskellHaddock //tests/haddock:haddock-lib-a failed (Exit 1) haddock_wrapper-haddock-lib-a failed: error executing command bazel-out/k8-dbg/bin/tests/haddock/haddock_wrapper-haddock-lib-a '--package-name=testsZShaddockZShaddock-lib-a' '--package-version=0' -D ... (remaining 96 argument(s) skipped)
...
tests/haddock/LibA.hs:10:1: error:
    Could not find module ‘Deep’
    There are files missing in the ‘haddock-lib-deep’ package,
    try running 'ghc-pkg check'.
    Use -v to see a list of the files searched for.
   |
10 | import Deep (deep_lib)
   | ^^^^^^^^^^^^^^^^^^^^^^
INFO: Elapsed time: 1.354s, Critical Path: 0.57s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```
This does not cause a CI failure on master, because we pass the flag `--build_tests_only` in profiling mode. This PR introduces a test case that depends on the haddock targets, which means they are now included despite `--build_tests_only`. We set the tag to disable the test on profiling builds.